backfill/2022-04-27-firefox-desktop-glean-rejected/README.md

```diff
@@ -0,0 +1,151 @@
+
+```bash
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_stable.baseline_v1 moz-fx-data-shared-prod:firefox_desktop_stable.baseline_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_stable.deletion_request_v1 moz-fx-data-shared-prod:firefox_desktop_stable.deletion_request_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_stable.events_v1 moz-fx-data-shared-prod:firefox_desktop_stable.events_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_stable.fog_validation_v1 moz-fx-data-shared-prod:firefox_desktop_stable.fog_validation_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_stable.metrics_v1 moz-fx-data-shared-prod:firefox_desktop_stable.metrics_v1
+```
+
+For the current day, we have to append into live tables instead:
+
+```bash
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_live.baseline_v1'$20220427' moz-fx-data-shared-prod:firefox_desktop_live.baseline_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_live.deletion_request_v1'$20220427' moz-fx-data-shared-prod:firefox_desktop_live.deletion_request_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_live.events_v1'$20220427' moz-fx-data-shared-prod:firefox_desktop_live.events_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_live.fog_validation_v1'$20220427' moz-fx-data-shared-prod:firefox_desktop_live.fog_validation_v1
+bq cp --append_table moz-fx-data-backfill-10:firefox_desktop_live.metrics_v1'$20220427' moz-fx-data-shared-prod:firefox_desktop_live.metrics_v1
+```
+
+## Step 5:  Clean up
+
+1.  DE will remove any GCS buckets created by Dataflow jobs:
+
+    ```bash
+    # BE CAREFUL!  This removes all GCS buckets from the target project.  It cannot be undone.
+
+    gsutil ls -p moz-fx-data-backfill-10 | xargs gsutil -m rm -r
+    ```
+
+    Log:
+
+    ```
+    ...
+    Operation completed over 250 objects.
+    Removing gs://dataflow-staging-us-central1-339436751542/...
+    ```
+
+2.  Data SRE will delete the errors for the payloads that have now been successfully ingested:
+
+```sql
+DELETE FROM `moz-fx-data-shared-prod.payload_bytes_error.structured`
+WHERE
+  DATE(submission_timestamp) BETWEEN '2022-04-12' AND '2022-04-27'
+  AND document_namespace = 'firefox-desktop'
+  AND error_message LIKE 'com.mozilla.telemetry.decoder.MessageScrubber$UnwantedDataException: 1684980'
+```
+
+This in practice produced an error message:
+
+> UPDATE or DELETE statement over table moz-fx-data-shared-prod.payload_bytes_error.structured would affect rows in the streaming buffer, which is not supported
+
+We modified this to not affect the current day (made the end date 2022-04-26) to get around the problem. This deleted 8,023,263 rows.
+
+We'll need to circle back to this after 2022-04-27 is complete to delete the final day:
+
+```sql
+DELETE FROM `moz-fx-data-shared-prod.payload_bytes_error.structured`
+WHERE
+  DATE(submission_timestamp) = '2022-04-27'
```

Comment:
If the initial copy step into `moz-fx-data-backfill-10.payload_bytes_error.backfill` was run before the end of the 2022-04-27 UTC day, maybe this should have an additional filter to avoid deleting all errors that happened after that point?
```suggestion
  DATE(submission_timestamp) = '2022-04-27'
  AND submission_timestamp <= (SELECT MAX(submission_timestamp) FROM `moz-fx-data-backfill-10.payload_bytes_error.backfill`)
```

Reply:
I think you've correctly identified that this overbroadly deleted data for the 27th. I came to the same conclusion late last night.

Since I already ran these commands I plan to use time travel to recover the deleted error pings to a temp table and then copy them back into pbe.structured. I'll post those queries here.

Reply:
Create the table:
```sql
CREATE TABLE `moz-fx-data-backfill-10.payload_bytes_error.backfill_fix`
 PARTITION BY DATE(submission_timestamp)
 CLUSTER BY
   submission_timestamp
 AS SELECT * FROM `moz-fx-data-backfill-10:payload_bytes_error.backfill` LIMIT 0
```

Populate the table:
```sql
SELECT *
FROM `payload_bytes_error.structured`
  FOR SYSTEM_TIME AS OF "2022-04-28 01:00:00.000000 UTC"
where date(submission_timestamp) = "2022-04-27" and submission_timestamp > (SELECT MAX(submission_timestamp) FROM `moz-fx-data-backfill-10.payload_bytes_error.backfill` where date(submission_timestamp) = "2022-04-27")
  AND document_namespace = 'firefox-desktop'
  AND error_message LIKE 'com.mozilla.telemetry.decoder.MessageScrubber$UnwantedDataException: 1684980'
```

I think this might technically still drop a few pings since we don't have guaranteed ordering on the streaming inserts, but I don't think we need to be worried about perfect fidelity here. If we are, it will be easier to reason by re-running the 27th in its entirety.

Append to prod table (not run yet, should add 11897 records to pbe.structured):

```
bq cp -a 'moz-fx-data-backfill-10:payload_bytes_error.backfill_fix' 'moz-fx-data-shared-prod:payload_bytes_error.structured'
```

@jklukas or @sean-rose if this looks correct I will run the final `bq cp`.

Reply:
I've run this bq cp.

Reply:
I've summarized this discussion in the text now.